quack/rmsnorm.py

```diff
@@ -632,9 +642,22 @@ def kernel(
                     tXsdW_other = cute.make_tensor(tXsdW.iterator + i * sdW.stride[0], tXsdW.layout)
                     cute.autovec_copy(tXsdW_other, tXrdW_other)
                     tXrdW.store(tXrdW.load() + tXrdW_other.load())
-                cute.copy(copy_atom_store_dW, tdWrdW, tdWgdW, pred=tdWpdW)
+                # Only convert for storing to weight.dtype if weights are not fp32
```

Comment:
`tdWgdW.element_type` should always be fp32, since we allocate dw_partial to be in fp32. 
You can just put an assert here that type is fp32.